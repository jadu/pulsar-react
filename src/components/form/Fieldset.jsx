import React from 'react';
import classnames from 'classnames';

export default class Fieldset extends React.Component {
  render() {
    let {
      children,
      legendClassName,
      legendText,
      ...props} = this.props;

    return (
      <fieldset {...props}>
          <legend className={legendClassName}>{legendText}</legend>
          {children}
      </fieldset>
    );
  }
}