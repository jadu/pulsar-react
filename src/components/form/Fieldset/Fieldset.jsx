import React from 'react';
import classnames from 'classnames';

export default class Fieldset extends React.Component {
  render() {
    let {
      children,
      legendText,
      ...props} = this.props;

    return (
      <fieldset>
          <legend>{legendText}</legend>
          {children}
      </fieldset>
    );
  }
}