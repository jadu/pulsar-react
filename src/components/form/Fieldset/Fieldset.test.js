'use strict';

import React from 'react';
import { render } from 'enzyme';

import Fieldset from './Fieldset';

it('Renders the fieldset without a legend', () => {
  const tree = render(
    <Fieldset>Foo</Fieldset>
  );
  expect(tree).toMatchSnapshot();
});

it('Renders the fieldset with the legend', () => {
  const tree = render(
    <Fieldset legendText="Bar">Foo</Fieldset>
  );
  expect(tree).toMatchSnapshot();
});

it('Renders the fieldset with the legend classes', () => {
  const tree = render(
    <Fieldset 
      legendClassName="baz"
      legendText="Bar"
    >
      Foo
    </Fieldset>
  );
  expect(tree).toMatchSnapshot();
});
