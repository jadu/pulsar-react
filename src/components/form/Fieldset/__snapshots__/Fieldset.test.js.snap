// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the fieldset with the legend 1`] = `
<fieldset>
  <legend>
    Bar
  </legend>
  Foo
</fieldset>
`;

exports[`Renders the fieldset with the legend classes 1`] = `
<fieldset>
  <legend
    class="baz"
  >
    Bar
  </legend>
  Foo
</fieldset>
`;

exports[`Renders the fieldset without a legend 1`] = `
<fieldset>
  Foo
</fieldset>
`;
